manual:  multi-index and single-index operators interactions

Octachron opened the pull request:
This PR adds a small paragraph after the presentation of multi-index operators to make it clearer that they are intended to be used in conjunction of single-index operators. Experimental data points indicate that this is not clear enough in the current version of the text. 

cc @mseri

mseri commented:
This looks good to me.

Even though it may seem obvious, and it is once you think about it, many people including myself got it wrong the first time.

hcarty commented on manual/manual/refman/exten.etex:
```diff
@@ -2390,7 +2390,25 @@ let syntax_compare vec mat t3 t4 =
+let (.%{;..}) = A.get
+let (.%{ }) a k = A.get a [|k|]
+\end{caml_example*}
+to handle uniformly both cases.
```
Maybe a small word order change: `to handle both cases uniformly`.

Octachron commented:
@gasche, would you mind casting a non-numerician eye on the wording (and eventually approving)?

gasche reviewed (approved):
My first impulse upon seeing that @mseri and @hcarty had reviewed was to approve on their behalf. @Octachron's "non-numerician" comment tricked me into believing there may be something specialized in the documentation that needed a more naive eye. This was just a trick to get an extra review: there is nothing weird in the new documentation comment and it looks just fine, so I'm approving on their behalf *and* mine.

gasche commented:
(you could fix the conflict and then merge in both trunk and 4.12.)

Octachron committed:
manual: warning for multi-index operators

This commit makes it clearer in the manual that multi-index operators
are intended to complement single-index ones.
